libcst/metadata/scope_provider.py

```diff
@@ -109,10 +125,20 @@ def __init__(self, name: str, scope: "Scope") -> None:
         self.__accesses = set()
 
     def record_access(self, access: Access) -> None:
-        self.__accesses.add(access)
+        if access.scope != self.scope or self._index < access._index:
+            self.__accesses.add(access)
 
     def record_accesses(self, accesses: Set[Access]) -> None:
-        self.__accesses |= accesses
+        later_accesses = {
+            access
+            for access in accesses
+            if access.scope != self.scope or self._index < access._index
+        }
+        self.__accesses |= later_accesses
+        earlier_accesses = accesses - later_accesses
+        if earlier_accesses != set() and self.scope.parent != self.scope:
+            for shadowed_assignment in self.scope.parent[self.name]:
+                shadowed_assignment.record_accesses(earlier_accesses)
```

Comment:
Why do we need this?

Reply:
NVM, I got it. It's to handle the accesses not part of later access. Add them to parent assignments make sense.

Reply:
Exactly!

Reply:
I used to have a comment here, it might've got lost during rebases. I'll add it back